swiftsimio/accelerated.py

```diff
@@ -168,3 +180,148 @@ def index_dataset(handle: Dataset, mask_array: np.array) -> np.array:
     ranges = ranges_from_array(mask_array)
 
     return read_ranges_from_file(handle, ranges, output_size, output_type)
+
+
+################################ ALEXEI: playing around with better read_ranges_from_file implementation #################################
+
+def get_chunk_ranges(ranges, chunk_size, array_length):
+    chunk_ranges = []
+    for bound in ranges:
+        lower = int(np.floor(bound[0]/chunk_size))*chunk_size
+        upper = min(int(np.ceil(bound[1]/chunk_size))*chunk_size, array_length)
+
+        if len(chunk_ranges) > 0:
+            assert(lower >= chunk_ranges[-1][0])
+            assert(upper >= chunk_ranges[-1][1])
+            if lower > chunk_ranges[-1][1]:
+                chunk_ranges.append([lower, upper])
+            elif lower <= chunk_ranges[-1][1]:
+                chunk_ranges[-1][1] = upper
+            else:
+                raise RuntimeError("computing chunk ranges has gone horribly wrong")
+        else:
+            chunk_ranges.append([lower, upper])
+
+    return chunk_ranges
+
+@jit(forceobj=True, fastmath=True)
+def expand_ranges(ranges):
+    length = 0 
+    for bounds in ranges:
+        length += bounds[1] - bounds[0]
```

Comment:
I don't really like this section of code... Can we replace it with something like `length = sum([x[1] - x[0] for x in ranges])`?